### PR TITLE
Add container uptime metric

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Add the `containerd.uptime` and `docker.uptime` metrics, two gauges that represents the uptime of a container (in seconds)

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by xornivore — DISMISSED** [review]
👍

File: pkg/collector/corechecks/containers/containerd.go [hunk 1]
```diff
@@ -296,6 +298,13 @@ func computeHugetlb(sender aggregator.Sender, huge []*cgroups.HugetlbStat, tags
 	}
 }
 
+func computeUptime(sender aggregator.Sender, ctn containers.Container, currentTime time.Time, tags []string) {
+	uptime := currentTime.Sub(ctn.CreatedAt).Seconds()
+	if uptime > 0 {
+		sender.Gauge("containerd.uptime", uptime, "", tags)
```
Comment (hkaj) [inline pkg/collector/corechecks/containers/containerd.go]: maybe call it containerd.container.uptime? Containerd.uptime makes me think of the uptime of containerd itself. Same for docker and cri

Comment (vboulineau) [reply pkg/collector/corechecks/containers/containerd.go]: I do agree, though it was to be consistent with the metrics we already have containerd.cpu, containerd.mem, etc. which are obviously not for containerd itself and should be called containerd.container.cpu IMHO.

File: pkg/collector/corechecks/containers/containerd.go [hunk 2]
```diff
@@ -177,6 +177,7 @@ func computeMetrics(sender aggregator.Sender, cu cutil.ContainerdItf, fil *ddCon
 		return
 	}
 
+	currentTime := time.Now()
```
Comment (hkaj) [inline pkg/collector/corechecks/containers/containerd.go]: probably a nitpick but setting currentTime before the loop, while calling `cu.Info()` in the loop might make the timing inaccurate. And in the cri check you get the time closer to the container uptime computation: https://github.com/DataDog/datadog-agent/pull/4753/files#diff-42a33bd390107e0c8b42f393b5946a9cR117 - we might want to harmonize that.

File: pkg/collector/corechecks/containers/docker.go [hunk 3]
```diff
@@ -175,6 +175,7 @@ func (d *DockerCheck) Run() error {
 	collectingContainerSizeDuringThisRun := d.instance.CollectContainerSize && d.collectContainerSizeCounter == 0
 
 	images := map[string]*containerPerImage{}
+	currentUnixTime := time.Now().Unix()
```
Comment (hkaj) [inline pkg/collector/corechecks/containers/docker.go]: same here, currentUnixTime is sampled before the loop and there are some potentially slow calls in the loop (i'm thinking of `GetStorageStats`, not sure if there are others)

File: pkg/collector/corechecks/containers/containerd.go (hunk 2 above)
Comment (vboulineau) [reply pkg/collector/corechecks/containers/containerd.go]: Yes you're right. I did it on purpose to be as accurate as possible, checking if calls trigger API calls or not, but in fact `cu.Info()` does trigger an API call, so we should get the time there.
Fixed.

File: releasenotes/notes/container-uptime-metrics-4f50973b086e74a3.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+features:
+  - |
+    Add docker.uptime and containerd.uptime metrics for all containers
```
Comment (hkaj) [inline releasenotes/notes/container-uptime-metrics-4f50973b086e74a3.yaml]:
```suggestion
    Add docker.uptime, cri.uptime, and containerd.uptime metrics for all containers
```

File: pkg/collector/corechecks/containers/docker.go (hunk 3 above)
Comment (vboulineau) [reply pkg/collector/corechecks/containers/docker.go]: Fixed

**Review by hkaj — DISMISSED** [review]

**Review by hkaj — APPROVED** [review]

**Pull request merged**